macros/src/parser/lifetimes.rs

```diff
@@ -0,0 +1,98 @@
+use quote::quote;
+use std::ops::Deref;
+use syn::{parse, spanned::Spanned, GenericArgument, Lifetime, PathArguments, Type};
+
+#[derive(Debug, Clone)]
+pub struct Lifetimes(Vec<Lifetime>);
+
+impl Lifetimes {
+    pub fn new() -> Lifetimes {
+        Lifetimes(Vec::new())
+    }
+
+    pub fn from_type(data_type: &Type) -> Result<Lifetimes, parse::Error> {
+        let mut lifetimes = Lifetimes::new();
+        get_lifetimes(data_type, &mut lifetimes)?;
+        Ok(lifetimes)
+    }
+
+    pub fn push(&mut self, lifetime: &Lifetime) {
+        if !self.0.contains(lifetime) {
+            self.0.push(lifetime.to_owned());
+        }
+    }
+
+    pub fn append(&mut self, lifetimes: &Lifetimes) {
+        for lifetime in lifetimes.0.iter() {
+            self.push(lifetime);
+        }
+    }
+
+    pub fn remove(&mut self, lifetimes: &Lifetimes) {
+        self.0.retain(|lt| !lifetimes.0.contains(lt));
+    }
```

Comment:
I think this is just https://doc.rust-lang.org/std/ops/trait.Sub.html?

Reply:
`Sub` is not implemented for `Vec`

Reply:
Ah, do you mean I could implement `Sub` for `Lifetimes`?

Reply:
I replaced `remove` with `Sub`